### PR TITLE
tidy fix for longer single line imports

### LINKED ISSUE #9864: Disable tidy length check for single import statements
If we have something like the following:
```rust
use dom::bindings::HorrendouslyLongClassNameThatWasMandatedByTheWHATWGSpec::HorrendouslyLongClassNameThatWasMandatedByTheWHATWGSpecBindings;
```

./mach test-tidy currently will fail because the line is longer than 120 characters. We should instead check whether the line is a use statement, and if so, only do a length check when there are braces detected in the use statement (otherwise don't perform any checks at all).

Code: `python/tidy.py`

### PR DESCRIPTION
**Pull request by creativcoder**
Fixes #9864

[Review on Reviewable](https://reviewable.io/reviews/servo/servo/9871)

### CODE REVIEW

File: python/tidy.py [hunk 1]
```diff
@@ -100,8 +100,10 @@ def check_length(file_name, idx, line):
     if file_name.endswith(".lock") or file_name.endswith(".json"):
         raise StopIteration
     max_length = 120
-    if len(line.rstrip('\n')) > max_length:
-        yield (idx + 1, "Line is longer than %d characters" % max_length)
+    current_line = line.rstrip('\n')
+    if 'use' in current_line and '{' in current_line or '}' in current_line:
```
Comment (canoon) [inline python/tidy.py]: Should this not be...
`not 'use' in current_line or '{' in current_line and '}' in current_line`

And maybe you want startswith 'use'

**Comment by Wafflespeanut** [comment]
-S-awaiting-review +S-needs-code-changes 

---

Reviewed 1 of 1 files at r1.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

*[python/tidy.py, line 104 \[r1\]](https://reviewable.io:443/reviews/servo/servo/9871#-KC07c7azZj9ioCjDDc4-r1-104:-KC0AdFVQkS3xdXbK1_f:364596388) ([raw file](https://github.com/servo/servo/blob/1d0e66b96fe6015a9a028a150c457e0a7282f4b1/python/tidy.py#L104)):*
This breaks tidy's length check. We should ignore the check only "only" for `use` statements that don't have the curly braces.

And yep, `startswith` would be very helpful here (for short-circuiting the loose cases)

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/servo/servo/9871#-:-KC09qMHBTMaPm8Pvkwj:1458266033)*

File: python/tidy.py (hunk 1 above)
Comment (creativcoder) [inline python/tidy.py]: ah, i get it. Thanks for correction. I'll make the changes.

**Comment by Wafflespeanut** [comment]
-S-awaiting-review +S-needs-code-changes 

---

Reviewed 1 of 1 files at r2.
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---

*[python/tidy.py, line 106 \[r2\]](https://reviewable.io:443/reviews/servo/servo/9871#-KC0PKKZghuDQ2-_fgMm:-KC0PKK_7HcFnb6RXDaA:1746889230) ([raw file](https://github.com/servo/servo/blob/de046be73c78e7f44344b7856dbe084bcb4d91f9/python/tidy.py#L106)):*
Um, this still doesn't work for non-use statements that don't have braces. Let's do something like,

```python
if len(current_line) > max_length:
    if '{' in current_line or not current_line.startswith('use'):
        yield ...
```

I think this would be more readable (and efficient, since there's no need to check for the prefix all the time). We (can, but we) shouldn't merge the `if` conditionals...

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/servo/servo/9871#-:-KC0QhOj6dtjlwipzi3j:1458266033)*

**Comment by Wafflespeanut** [comment]
@creativcoder It's been suggested that we could go for an alternative solution. So, we'll need to make one last modification. Currently, tidy doesn't care very much about the `use` statements - it just emits an error [if it doesn't find a semicolon at EOL](https://github.com/servo/servo/blob/a9c8f592a9521376678187c6917449e3dd48c373/python/tidy.py#L369). Now, we need to constrain it a little bit, and emit the error only if we find a group of imports in the line (i.e, a curly brace).

I'm sorry I didn't think about the issue very well at first :)

**Comment by Wafflespeanut** [comment]
@bors-servo r+

Thanks! :)

**Pull request merged**